src/checkers/inference/solver/frontend/ConstraintSerializer.java

```diff
@@ -34,13 +35,15 @@
     private Serializer<S, T> realSerializer;
 
     @SuppressWarnings("unchecked")
-    public ConstraintSerializer(String backEndType, Lattice lattice) {
+    public ConstraintSerializer(BackEndType backEndType, Lattice lattice) {
         try {
-            if (backEndType.equals(Constants.LINGELING)) {
+            String backEndPath = backEndType.getFullyQualifiedName();
+            if (backEndType.getBackEndType().equals(BackEndTypeEnum.Lingeling)) {
```

Comment:
Thanks for the explanation! Now I understand why Serializer and BackEnd are highly decoupled.